тестовое задание 4 спринт

vasilevaoi opened the pull request.

vasilevaoi committed:
тестовое задание 4 спринт

mmbabaev commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,72 +1,167 @@
+    @IBOutlet private var ButtonYes: UIButton!
+    @IBOutlet private var imageView: UIImageView!
+    @IBOutlet private var textLabel: UILabel!
+    @IBOutlet weak var counterLabel: UILabel!
```
пропущен private

mmbabaev commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,72 +1,167 @@
 import UIKit
 
 final class MovieQuizViewController: UIViewController {
+    @IBOutlet private var ButtonNo: UIButton!
```
названия переменных принято начинать с маленькой буквы

mmbabaev commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,72 +1,167 @@
+    @IBAction func yesButtonClicked(_ sender: UIButton) {
+        let currentQuestion = questions[currentQuestionIndex]
+        let givenAnswer = true
+        showAnswerResult(isCorrect: givenAnswer == currentQuestion.correctAnswer) // 3
```
лучше убирать комментарии из учебника и не копировать код, а перепечатывать его самостоятельно. На начальных этапах важно "набить руку"

vasilevaoi committed:
исправлены критические замечания
добавлено блокирование кнопок на время выполнения операции запроса следующего вопроса

vasilevaoi committed:
поправлена структура кода

vasilevaoi committed:
рефакторинг кода
